dspace-api/src/main/java/org/dspace/app/harvest/Harvest.java

```diff
@@ -36,221 +33,224 @@
+            runHarvest(context, collection);
         } else if ("start".equals(command)) {
             // start the harvest loop
             startHarvester();
         } else if ("reset".equals(command)) {
             // reset harvesting status
-            resetHarvesting();
+            resetHarvesting(context);
         } else if ("purgeAll".equals(command)) {
             // purge all collections that are set up for harvesting (obviously for testing purposes only)
-            if (eperson == null) {
-                System.out
-                    .println("Error - an eperson must be provided");
-                System.out.println(" (run with -h flag for details)");
-                System.exit(1);
+            if (context.getCurrentUser() == null) {
+                handler.logError("An eperson must be provided (run with -h flag for details)");
+                throw new UnsupportedOperationException("An eperson must be provided");
             }
 
             List<HarvestedCollection> harvestedCollections = harvestedCollectionService.findAll(context);
             for (HarvestedCollection harvestedCollection : harvestedCollections) {
-                System.out.println(
-                    "Purging the following collections (deleting items and resetting harvest status): " +
-                        harvestedCollection
-                            .getCollection().getID().toString());
-                harvester.purgeCollection(harvestedCollection.getCollection().getID().toString(), eperson);
+                handler.logInfo(
+                        "Purging the following collections (deleting items and resetting harvest status): " +
+                                harvestedCollection
+                                        .getCollection().getID().toString());
+                purgeCollection(context, harvestedCollection.getCollection().getID().toString());
             }
             context.complete();
         } else if ("purge".equals(command)) {
             // Delete all items in a collection. Useful for testing fresh harvests.
-            if (collection == null || eperson == null) {
-                System.out
-                    .println("Error - a target collection and eperson must be provided");
-                System.out.println(" (run with -h flag for details)");
-                System.exit(1);
+            if (collection == null || context.getCurrentUser() == null) {
+                handler.logError("A target collection and eperson must be provided (run with -h flag for details)");
+                throw new UnsupportedOperationException("A target collection and eperson must be provided");
             }
 
-            harvester.purgeCollection(collection, eperson);
+            purgeCollection(context, collection);
             context.complete();
 
             //TODO: implement this... remove all items and remember to unset "last-harvested" settings
+        } else if ("reimport".equals(command)) {
```

Comment:
Can the TODO above this line be removed now?  It looks like you've implemented "reimport"